### DISCUSSION TITLE
Question about increase IVF_SQ8's memory

### DISCUSSION THREAD

**Discussion by hjlee9182** [milvus-io/milvus#31314] [Q&A and General discussion]
Milvus 2.3.11
1 query node

Whne I search & query for collection with IVF_SQ8. ( Collection, IVF_SQ8, 768dim, 1m rows)
That query node memory is increasing.
Every data is sealed.
Because I know that if I use IVF_SQ8 index, all indexes load into memory at the beginning, but I don't know why memory keeps increasing.
IVF_FLAT works as expected, so the memory does not stretch after first loading the memory.
Can you Explain this situation..?

<img width="1211" alt="image" src="https://github.com/milvus-io/milvus/assets/46337218/2a612ad4-8213-4271-bb07-64d8394f5c89">
<img width="1807" alt="image" src="https://github.com/milvus-io/milvus/assets/46337218/3abb6198-ff7e-49d3-b4fd-abe432172682">
<img width="1362" alt="image" src="https://github.com/milvus-io/milvus/assets/46337218/cb6755e5-cca8-43a1-b23f-98b610d5e9f2">
<img width="612" alt="image" src="https://github.com/milvus-io/milvus/assets/46337218/c9d6bae6-ca22-4ac2-a120-9e181ab6177d">

**Comment by yiwangdr** [comment]
1. query and search consumes memory too. If you pause search/query, do the memory decrease back? 
2. If not, could you please pprof the heap? 
`curl http://{address_of_your query_node}/debug/pprof/heap > ~/tmp/heap.out`

**Reply by hjlee9182** [reply]
1. No, the memory is reamin. After release collection memory is decreases.
2. How can i get pprof the heap inside query node pod?

**Reply by yiwangdr** [reply]
`curl http://localhost:9091/debug/pprof/heap > ~/tmp/heap.out`

Please upload heap.out. We will take a look.

**Reply by hjlee9182** [reply]
Okay, here!
[heap.out.zip](https://github.com/milvus-io/milvus/files/14622992/heap.out.zip)

**Reply by yiwangdr** [reply]
The numbers look surprisingly low. 
What was the memory usage when you run the curl command?

**Reply by hjlee9182** [reply]
Memory usage is 5.82GB.
This is start from 3.04GB.
1. after load 3.04GB
2. after query & search 5.82GB
3. curl command...

**Reply by xiaofan-luan** [reply]
as I said, all memory consumed by memory allocator.  it always try to eat up memory.
if you want to use less memories, simply change your pod limit to 4g and I guess it won't OOM either

**Reply by hjlee9182** [reply]
@xiaofan-luan 
I don't know much about Memory allocator, so if I ask you a question, IVF_FLAT didn't increase memory even when I did query&search, is it because I already allocated enough memory when loading?

**Comment by xiaofan-luan** [comment]
i guess the memory is mostly eat up by memory allocater. 

Memory allocator try to allocate more memory and reserved. From your case it seems that I won't increase.

BTW, if you only have 1 million 768 dimension datas, IVF_FLAT could fit as well into 8g memory or even 4g

**Reply by hjlee9182** [reply]
Umm.. I don't understand what you mena. Is memory allocator include milvus?

**Reply by xiaofan-luan** [reply]
nope , by saying memory allocator I mean jemalloc or glibc.

**Reply by xiaofan-luan** [reply]
by default milvus use jemalloc, and there are maybe some parameters to tune
